src/script/components/list/ConversationListCallingCell.test.ts

```diff
@@ -0,0 +1,129 @@
+    super(ConversationListCallingCell, props);
+  }
+
+  getAcceptButton = () => this.get('[data-uie-name="do-call-controls-call-accept"]');
+  getDeclineButton = () => this.get('[data-uie-name="do-call-controls-call-decline"]');
+  getOutgoingLabel = () => this.get('[data-uie-name="call-label-outgoing"]');
+  getConnectingLabel = () => this.get('[data-uie-name="call-label-connecting"]');
+  getCallDuration = () => this.get('[data-uie-name="call-duration"]');
+}
+
+const createCall = (state: CALL_STATE, selfUser = new User(createRandomUuid()), selfClientId = createRandomUuid()) => {
+  const selfParticipant = new Participant(selfUser, selfClientId);
+  const call = new Call('', '', 0, selfParticipant, CALL_TYPE.NORMAL, {
+    currentAvailableDeviceId: {
+      audioOutput: ko.pureComputed(() => 'test'),
+    },
+  } as MediaDevicesHandler);
+  call.state(state);
+  return call;
+};
+
+const createProps = async () => {
+  const testFactory = new TestFactory();
+
+  await testFactory.exposeCallingActors();
+  testFactory.calling_repository['callState'].isMuted(false);
+  jest.spyOn(testFactory.calling_repository, 'supportsScreenSharing', 'get').mockReturnValue(true);
+
+  await testFactory.exposeTeamActors();
+  jest.spyOn(testFactory.team_repository, 'isExternal').mockImplementation(() => false);
+
+  const conversation = new Conversation();
+  conversation.participating_user_ets([new User('id')]);
+  return {
+    call: createCall(CALL_STATE.MEDIA_ESTAB),
+    callActions: {} as CallActions,
+    callingRepository: testFactory.calling_repository,
+    conversation,
+    hasAccessToCamera: true,
+    isSelfVerified: true,
+    maximizedTileVideoParticipant: undefined,
+    multitasking: {isMinimized: ko.observable(false)},
+    teamRepository: testFactory.team_repository,
+    videoGrid: {grid: [], thumbnail: undefined},
+    videoSpeakersActiveTab: VideoSpeakersTab.ALL,
+  } as CallingCellProps;
+};
+
+describe('ConversationListCallingCell', () => {
+  it('displays an incoming ringing call', async () => {
+    const props = await createProps();
+    props.call.state(CALL_STATE.INCOMING);
+    const callingCellPage = new ConversationListCallingCellPage(props);
+
+    expect(callingCellPage.getAcceptButton().exists()).toBe(true);
+    expect(callingCellPage.getDeclineButton().exists()).toBe(true);
+  });
+  it('displays an outgoing ringing call', async () => {
+    const props = await createProps();
+    props.call.state(CALL_STATE.OUTGOING);
+    const callingCellPage = new ConversationListCallingCellPage(props);
+
+    expect(callingCellPage.getOutgoingLabel().exists()).toBe(true);
+  });
+  it('displays a call that is connecting', async () => {
+    const props = await createProps();
+    props.call.state(CALL_STATE.ANSWERED);
+    const callingCellPage = new ConversationListCallingCellPage(props);
+
+    expect(callingCellPage.getConnectingLabel().exists()).toBe(true);
+  });
+  it('displays the running time of an ongoing call', async () => {
```

Comment:
please add a line break after `it` blocks

Reply:
I'm against adding an extra line break after `it` blocks. We never do that.